modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderLocalServiceTest.java

```diff
@@ -0,0 +1,329 @@
+
+			Assert.assertNotNull(
+				_objectFolderLocalService.getObjectFolder(
+					company.getCompanyId(),
+					ObjectFolderConstants.NAME_UNCATEGORIZED));
+
+			objectFolder = _addObjectFolder(true, user);
+		}
+		finally {
+			_companyLocalService.deleteCompany(company);
+
+			PermissionThreadLocal.setPermissionChecker(
+				originalPermissionChecker);
+			PrincipalThreadLocal.setName(originalName);
+		}
+
+		Assert.assertNull(
+			_objectFolderLocalService.fetchObjectFolder(
+				company.getCompanyId(),
+				ObjectFolderConstants.NAME_UNCATEGORIZED));
+
+		Assert.assertNull(
+			_objectFolderLocalService.fetchObjectFolder(
+				objectFolder.getObjectFolderId()));
+	}
+
+	@Test
+	public void testDeleteObjectFolder() throws Exception {
+		AssertUtils.assertFailure(
+			UnsupportedOperationException.class,
+			"Uncategorized cannot be deleted",
+			() -> _objectFolderLocalService.deleteObjectFolder(
+				_uncategorizedObjectFolder.getObjectFolderId()));
+
+		ObjectFolder objectFolder = _addObjectFolder(
+			true, TestPropsValues.getUser());
+
+		int countUncategorizedObjectFolder =
+			_objectDefinitionLocalService.getObjectDefinitions(
+				_uncategorizedObjectFolder.getObjectFolderId()
+			).size();
+
+		objectFolder = _objectFolderLocalService.deleteObjectFolder(
+			objectFolder.getObjectFolderId());
+
+		Assert.assertNull(
+			_objectFolderLocalService.fetchObjectFolder(
+				objectFolder.getObjectFolderId()));
+
+		Assert.assertEquals(
+			0,
+			_objectDefinitionLocalService.getObjectDefinitions(
+				objectFolder.getObjectFolderId()
+			).size());
+
+		Assert.assertEquals(
+			countUncategorizedObjectFolder + 2,
+			_objectDefinitionLocalService.getObjectDefinitions(
+				_uncategorizedObjectFolder.getObjectFolderId()
+			).size());
+	}
+
+	@Test
+	public void testUpdateObjectFolder() throws Exception {
+		ObjectFolder objectFolder1 = _addObjectFolder(
+			false, TestPropsValues.getUser());
+
+		AssertUtils.assertFailure(
+			DuplicateObjectFolderExternalReferenceCodeException.class,
+			StringBundler.concat(
+				"Duplicate object folder with external reference code ",
+				_uncategorizedObjectFolder.getExternalReferenceCode(),
+				" and company ", _uncategorizedObjectFolder.getCompanyId()),
+			() -> _objectFolderLocalService.updateObjectFolder(
+				_uncategorizedObjectFolder.getExternalReferenceCode(),
+				objectFolder1.getObjectFolderId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				Collections.emptyList()));
+
+		AssertUtils.assertFailure(
+			ObjectFolderLabelException.class,
+			"Label is null for locale " + LocaleUtil.US.getDisplayName(),
+			() -> _objectFolderLocalService.updateObjectFolder(
+				RandomTestUtil.randomString(),
+				objectFolder1.getObjectFolderId(), null,
+				Collections.emptyList()));
+
+		_objectFolderLocalService.deleteObjectFolder(objectFolder1);
+
+		ObjectFolder objectFolder2 = _addObjectFolder(
+			false, TestPropsValues.getUser());
```

Comment:
Why did you need to delete and add again?

Reply:
Can't you go ahead and update the variable? Do you need to delete and add again?